tinytuya/core.py

```diff
@@ -1654,6 +1654,7 @@ def add_dps_to_request(self, dp_indicies):
             self.dps_to_request.update({str(index): None for index in dp_indicies})
 
     def set_version(self, version): # pylint: disable=W0621
+        version = float(version)

```

Comment:
Good call! Yes, this surfaces as an issue every 3-6mo or so.

Reply:
Yeah, I thought I'd sneak that one in there :)  I'm not marking it as 'closes #507' as I think there's more we can do to type-safe everything.